### ISSUE TITLE
 TLS 1.3 ffdhe invalid parameters (FIPS)

### ISSUE THREAD

**Issue by nmav** [tlsfuzzer/tlsfuzzer#441]
* generate key shares not using the generator associated with the prime (not from the generator's subgroup), verify that server aborts the connection with `illegal_parameter`

do the tests for all the ffdhe groups.

**Comment by tomato42** [comment]
@nmav I've rephrased the entry a bit, is this what you had in mind?

**Comment by nmav** [comment]
Hmmm, we are getting into implementation details. Not sure if it is accurate. You could find the generator of the other subgroup and do what it says above, or you could generate random pubkeys until you find one that does not belong to the subgroup (test y^((p-1)/2) mod p != 1).
[as in 5.6.2.1 of https://csrc.nist.gov/publications/detail/sp/800-56a/rev-3/final )

**Comment by tomato42** [comment]
yes, how we get those invalid key shares is secondary (but will help whoever picks this up), the part I was asking about primarily was the server abort with alert
